### ISSUE TITLE
Cannot start tkinter-based example on Python 3.5.1 using Mac Homebrew for Python and Tk

### ISSUE THREAD

**Issue by josePhoenix** [matplotlib/matplotlib#5837]
Potentially related to #5143 or #2025?

Using Homebrew on OS X 10.10.5, i installed `homebrew/dupes/tcl-tk`, uninstalled and reinstalled `python3`, then tried to rebuild matplotlib against the newer Python and Tk with `pip3 install --no-binary :all: matplotlib`. (Note that since I do not have root access to this computer, my Homebrew installation is under `$HOME`.)

When I try to start the `embedding_in_tk.py` [example](http://matplotlib.org/examples/user_interfaces/embedding_in_tk.html), I get the following errors:

```
$ python3 embedding_in_tk.py
objc[5671]: Class TKApplication is implemented in both /Users/jlong/opt/tcl-tk/lib/libtk8.6.dylib and /System/Library/Frameworks/Tk.framework/Versions/8.5/Tk. One of the two will be used. Which one is undefined.
objc[5671]: Class TKMenu is implemented in both /Users/jlong/opt/tcl-tk/lib/libtk8.6.dylib and /System/Library/Frameworks/Tk.framework/Versions/8.5/Tk. One of the two will be used. Which one is undefined.
objc[5671]: Class TKContentView is implemented in both /Users/jlong/opt/tcl-tk/lib/libtk8.6.dylib and /System/Library/Frameworks/Tk.framework/Versions/8.5/Tk. One of the two will be used. Which one is undefined.
objc[5671]: Class TKWindow is implemented in both /Users/jlong/opt/tcl-tk/lib/libtk8.6.dylib and /System/Library/Frameworks/Tk.framework/Versions/8.5/Tk. One of the two will be used. Which one is undefined.
Traceback (most recent call last):
  File "/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/tkagg.py", line 22, in blit
    id(data), colormode, id(bbox_array))
_tkinter.TclError: invalid command name "PyAggImagePhoto"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "embedding_in_tk.py", line 34, in <module>
    canvas.show()
  File "/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/backend_tkagg.py", line 355, in draw
    tkagg.blit(self._tkphoto, self.renderer._renderer, colormode=2)
  File "/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/tkagg.py", line 30, in blit
    id(data), colormode, id(bbox_array))
_tkinter.TclError
```

Running `otool -L` on the compiled `_tkagg` extension shows it using the system's Tcl/Tk instead of Homebrew's:

```
$ otool -L /Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/_tkagg.cpython-35m-darwin.so
/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/_tkagg.cpython-35m-darwin.so:
    /System/Library/Frameworks/Tcl.framework/Versions/8.5/Tcl (compatibility version 8.5.0, current version 8.5.9)
    /System/Library/Frameworks/Tk.framework/Versions/8.5/Tk (compatibility version 8.5.0, current version 8.5.9)
    /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 120.0.0)
    /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
```

So, my question is, am I missing a necessary flag to install matplotlib from source on a Mac with this particular combination of circumstances? Is there a way to tell it to use the Homebrew Tcl/Tk?

(Oddly enough, I'm only building my own Tcl/Tk to solve _another_ issue: http://sourceforge.net/p/tktoolkit/bugs/2966/ http://bugs.python.org/issue10731)

**Comment by tacaswell** [comment]
That is a sign your compilation did not go correctly.  Can you check if the `PyAggImagePhoto` is in `_tkagg.cpython-35m-darwin.so`?

**Comment by josePhoenix** [comment]
That doesn't appear to be an exported symbol in `_tkagg.cpython-35m-darwin.so`:

```
10:26:01 tanuki:~ jlong$ nm -g /Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/_tkagg.cpython-35m-darwin.so | grep Py
                 U _PyArg_ParseTuple
                 U _PyBytes_AsString
                 U _PyCapsule_GetPointer
                 U _PyCapsule_Type
                 U _PyErr_Clear
                 U _PyErr_Format
                 U _PyErr_Occurred
                 U _PyErr_Print
                 U _PyErr_SetString
                 U _PyExc_AttributeError
                 U _PyExc_ImportError
                 U _PyExc_RuntimeError
                 U _PyExc_TypeError
                 U _PyExc_ValueError
                 U _PyFloat_AsDouble
                 U _PyImport_ImportModule
00000000000034f0 T _PyInit__tkagg
                 U _PyModule_Create2
                 U _PyObject_CallMethod
                 U _PyObject_GetAttrString
                 U _PyObject_HasAttrString
                 U _PyObject_IsTrue
                 U _PySequence_Check
                 U _PySequence_GetItem
                 U _PySequence_Size
                 U _PyUnicode_AsASCIIString
                 U _Py_BuildValue
                 U __Py_NoneStruct
10:26:06 tanuki:~ jlong$ otool -TV /Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/_tkagg.cpython-35m-darwin.so
/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/matplotlib/backends/_tkagg.cpython-35m-darwin.so:
Table of contents (0 entries)
module name      symbol name
```

**Comment by josePhoenix** [comment]
I cloned v1.5.1 and did `python3 setup.py build -f`. I captured the whole build log and can post somewhere, but here's the lines building the `_tkagg` extension:

```
building 'matplotlib.backends._tkagg' extension
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib_backends__tkagg_ARRAY_API -DN
PY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/numpy/core/include -I/us
r/local/include -I/usr/include -I/usr/X11/include -I/opt/X11/include -I. -I/System/Library/Frameworks/Tcl.framework/Headers -I/System/Library/Frameworks/Tcl.framework/Versions/Cur
rent/PrivateHeaders -I/System/Library/Frameworks/Tk.framework/Headers -I/System/Library/Frameworks/Tk.framework/Versions/Current/PrivateHeaders -Iextern/agg24-svn/include -I/Users
/jlong/include -I/Users/jlong/opt/openssl/include -I/Users/jlong/opt/sqlite/include -I/Users/jlong/opt/tcl-tk/include -I/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Pytho
n.framework/Versions/3.5/include/python3.5m -c src/py_converters.cpp -o build/temp.macosx-10.10-x86_64-3.5/src/py_converters.o -framework Tcl -framework Tk
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib_backends__tkagg_ARRAY_API -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/numpy/core/include -I/usr/local/include -I/usr/include -I/usr/X11/include -I/opt/X11/include -I. -I/System/Library/Frameworks/Tcl.framework/Headers -I/System/Library/Frameworks/Tcl.framework/Versions/Current/PrivateHeaders -I/System/Library/Frameworks/Tk.framework/Headers -I/System/Library/Frameworks/Tk.framework/Versions/Current/PrivateHeaders -Iextern/agg24-svn/include -I/Users/jlong/include -I/Users/jlong/opt/openssl/include -I/Users/jlong/opt/sqlite/include -I/Users/jlong/opt/tcl-tk/include -I/Users/jlong/homebrew/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/include/python3.5m -c src/_tkagg.cpp -o build/temp.macosx-10.10-x86_64-3.5/src/_tkagg.o -framework Tcl -framework Tk
clang++ -bundle -undefined dynamic_lookup -L/Users/jlong/opt/tcl-tk/lib build/temp.macosx-10.10-x86_64-3.5/src/py_converters.o build/temp.macosx-10.10-x86_64-3.5/src/_tkagg.o -L/usr/local/lib -L/usr/lib -L/usr/X11/lib -L/opt/X11/lib -L/Users/jlong/lib -L/Users/jlong/opt/openssl/lib -L/Users/jlong/opt/sqlite/lib -L/Users/jlong/opt/tcl-tk/lib -o build/lib.macosx-10.10-x86_64-3.5/matplotlib/backends/_tkagg.cpython-35m-darwin.so -framework Tcl -framework Tk
```

**Comment by Laspimon** [comment]
Any luck fixing this? I'm on OSX 10.10.5 installing through Macports, and I seem to have the same problem.

**Comment by OliverUrbann** [comment]
Same problem here with Macports on OSX 10.11.

**Comment by efiring** [comment]
Duplicate of #2025

**Comment by tacaswell** [comment]
This may be fixed in a more through way by #6442 .

**Comment by dstansby** [comment]
Since this now seems to be working fine for me, and the duplicate bug is closed I'm going to close this - if anyone still has the issue let me know though and I'll re-open.
